typescript/demo/function-call-stream.ts

```diff
@@ -217,12 +222,20 @@ async function functionCallingWithAIConfig() {
       return;
     }
 
-    const rawResponse = output.metadata?.rawResponse;
-    const functionCall = rawResponse?.function_call;
+    let functionCall: FunctionData | null = null;
+    if (output.data?.hasOwnProperty("value")) {
+      const outputData = output.data as OutputDataWithValue;
+      if (outputData.kind === "tool_calls") {
+        outputData as OutputDataWithToolCallsValue;
+        // Typescript schema does not support array of function calls yet,
+        // but Python does so doing this for forward compatibility
+        functionCall = outputData.value[outputData.value.length - 1].function;
+      }
+    }
```

Comment:
nit: this can be simplified:

```
if (typeof output.data !== "string" && output.data.kind === "tool_calls") {
  // OutputDataWithToolCallsValue type should be automatically inferred by the TS compiler in this if block
  const outputData: OutputDataWithToolCallsValue = output.data
  if (Array.isArray(outputData)) {
    functionCall = outputData.value[outputData.value.length - 1].function;
  } else {
    functionCall = outputData.value.function;
  }
}
```

Reply:
> if (Array.isArray(outputData)) { functionCall = outputData.value[outputData.value.length - 1].function; } else { functionCall = outputData.value.function; }

We already set in our schema that `outputData.value.function` must always be an array (which follows what OpenAI has defined). But yea I can do this for future-proofing if you like

Reply:
Ah, got it. In that case we can ignore the Array.isArray check. Thanks for confirming. Rest of the suggested refactor is still valid

Reply:
Also I don't think this works because we're supporting arbitrary JSON types, so it's not guaranteed to have `output.data.kind` defined:
<img width="1010" alt="Screenshot 2023-12-27 at 14 56 15" src="https://github.com/lastmile-ai/aiconfig/assets/151060367/91b38cf6-71de-4040-92cb-a08ef81a3693">